### ISSUE TITLE
Check for nested dynamic arrays in TypeChecker

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#2764]
Encoding:
```
contract C {
  function f() returns (uint[][] memory) {
  }
}
```

Results in:
```
Unimplemented feature:
/Users/alex/Projects/solidity/libsolidity/codegen/ArrayUtils.cpp(290): Throw in function void dev::solidity::ArrayUtils::copyArrayToMemory(const dev::solidity::ArrayType &, bool) const
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::UnimplementedFeatureError>
std::exception::what: Nested dynamic arrays not implemented here.
[dev::tag_comment*] = Nested dynamic arrays not implemented here.
```

Decoding:
```
contract C {
  function f(uint[][]) {
  }
}
```

Results in:
```
Unimplemented feature:
/Users/alex/Projects/solidity/libsolidity/codegen/ContractCompiler.cpp(333): Throw in function void dev::solidity::ContractCompiler::appendCalldataUnpacker(const TypePointers &, bool)
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::UnimplementedFeatureError>
std::exception::what: Nested arrays not yet implemented.
[dev::tag_comment*] = Nested arrays not yet implemented.
```

**Comment by axic** [comment]
`ABIEncoderV2` can encode it, hoping it will move from experimental mode soon, this will not be needed.

**Comment by chriseth** [comment]
This has been fixed by #3549
